repose-aggregator/components/services/container-configuration-service/impl/src/main/scala/org/openrepose/nodeservice/containerconfiguration/ContainerConfigurationServiceImpl.scala

```diff
@@ -63,9 +63,29 @@ class ContainerConfigurationServiceImpl @Inject()(@Value(ReposeSpringProperties.
     configurationService.unsubscribeFrom(ContainerConfigurationFilename, this)
   }
 
-  override def getVia: Optional[String] = {
+  override def getRequestVia: Optional[String] = {
     initializationCheck()
-    Optional.ofNullable(patchedDeploymentConfiguration.getVia)
+    Option(patchedDeploymentConfiguration.getViaHeader)
+      .map(viaHdr => Optional.ofNullable(viaHdr.getRequestPrefix))
+      // TODO for v9.0.0.0: This will need updated.
+      .getOrElse(Optional.ofNullable(patchedDeploymentConfiguration.getVia))
+      //.getOrElse(Optional.empty())
+  }
+
+  override def getResponseVia: Optional[String] = {
+    initializationCheck()
+    Option(patchedDeploymentConfiguration.getViaHeader)
+      .map(viaHdr => Optional.ofNullable(viaHdr.getResponsePrefix))
+      // TODO for v9.0.0.0: This will need updated.
+      .getOrElse(Optional.ofNullable(patchedDeploymentConfiguration.getVia))
+      //.getOrElse(Optional.empty())
+  }
+
+  override def includeViaReposeVersion: Boolean = {
+    initializationCheck()
+    Option(patchedDeploymentConfiguration.getViaHeader)
+      .map(_.isReposeVersion)
+      .orElse(Option(true)).get
```

Comment:
IntelliJ had even a better suggestion to take it to a `.forall(_.isReposeVersion)` after making the `.getOrElse(true)` update.